src/generic/graphicc.cpp

```diff
@@ -1124,15 +1130,22 @@ wxCairoFontData::InitFontComponents(const wxString& facename,
 }
 
 wxCairoFontData::wxCairoFontData( wxGraphicsRenderer* renderer, const wxFont &font,
-                         const wxColour& col )
+                                  const wxRealPoint& dpi, const wxColour& col )
     : wxGraphicsObjectRefData(renderer)
 #ifdef __WXGTK__
     , m_wxfont(font)
 #endif
 {
     InitColour(col);
 
+#ifndef __WXMAC__
+    m_size = !dpi.y
+        ? double(font.GetPixelSize().GetHeight())
+        : double(font.GetFractionalPointSize() * dpi.y / 72);
```

Comment:
Could you please explain why do we do this and why do we do a different thing under Mac?

This must be hiding some more fundamental incompatibility/difference between the platforms, but I don't know which one.

Reply:
I think there are two differences here between macOS and Windows.

The first one is that macOS is based on 72dpi and Windows on 96dpi. Fonts are in 72ppi, so on Windows they have to be corrected to the DPI. On macOS no correction is needed.

Second, on Windows the font has to be adjusted to the screen DPI (96 at 100%, 192 at 200%) to get the correct font size. On macOS this is not needed (content scale stuff?) so you basically end up with `font.GetFractionalPointSize() * 72 / 72)`.

Reply:
I'd prefer to use `wxDisplay::GetStdPPIValue() / 72`, this should be optimized away under Mac and would result in more clear code IMO.

As for "DPI correction", I'd indeed expect it to be taken care of by `GetContentScaleFactor()`, but its value is the same for GTK and macOS, while this code is Mac-specific. I.e. it's not so much the difference between MSW and Mac that I don't understand, but the difference between Mac and GTK.

Reply:
I tried both on GTK and the font looks exactly the same. Looking further into this, it is because GTK uses pango for font drawing. `m_size` isn't even used, it uses `m_wxfont`.

Reply:
I've restructured `wxCairoFontData` constructor and added some comments, to address this and above comments.